### PR TITLE
add MikMod_update that sound works and fix compiling

### PR DESCRIPTION
**Pull request by HybridDog**

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 static const int   WIDTH=60;
 static const int   ANGLE=4;
-static const int   DELAY=50000;
+static const int   DELAY=20000;
```
Comment (HybridDog) [inline src/main.c:11]: 50000 ms delay was too slow which causes the sound to be played weird

File: src/main.c [hunk 2]
```diff
@@ -50,7 +50,8 @@ int main() {
   // Extract module song payload
   char fname[] = "/tmp/file-XXXXXX";
   int fd = mkstemp(fname);
-  write(fd, src_music_xm, src_music_xm_len);
+  if (write(fd, src_music_xm, src_music_xm_len) < 0)
+    return -1;
```
Comment (HybridDog) [inline src/main.c:54]: l did it to fix the unused result message.

File: src/main.c [hunk 3]
```diff
@@ -66,6 +67,7 @@ int main() {
 
   // Animation
   while (1) {
+    MikMod_Update();
```
Comment (HybridDog) [inline src/main.c:70]: This seems to be necessary.
http://web.mit.edu/outland/src/libmikmod-3.1.9/docs/mikmod.html#TOC_67

File: compile.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+autoconf &&\
+./configure &&\
+sed -i 's/main.c/main.c `libmikmod-config --cflags` `libmikmod-config --libs`/g' Makefile &&\
```
Comment (HybridDog) [inline compile.sh:3]: maybe there's a better way than editing Makefile

**Comment by djui** [comment]
Interesting findings, thanks!

Not too happy with the `compile.sh` but if it works for you currently we can keep it and revisit later.

:+1:

**Pull request merged**